runtime/lua/vim/health.lua

```diff
@@ -4,22 +4,140 @@ function M.report_start(msg)
+    return true
+  end
+  return false
+end
+
+-- Handler for s:system() function.
+local function system_handler(self, _, data, event)
+  if event == 'stderr' then
+    if self.add_stderr_to_output then
+      self.output = self.output .. table.concat(data, '')
+    else
+      self.stderr = self.stderr .. table.concat(data, '')
+    end
+  elseif event == 'stdout' then
+    self.output = self.output .. table.concat(data, '')
+  end
+end
+
+-- Attempts to construct a shell command from an args list.
+-- Only for display, to help users debug a failed command.
+local function shellify(cmd)
+  if type(cmd) ~= 'table' then
+    return cmd
+  end
+  local escaped = {}
+  for i, v in ipairs(cmd) do
+    if v:match('[^A-Za-z_/.-]') then
+      escaped[i] = vim.fn.shellescape(v)
+    else
+      escaped[i] = v
+    end
+  end
+  return table.concat(escaped, ' ')
+end
+
+function M.cmd_ok(cmd)
+  local out = vim.fn.system(cmd)
+  return vim.v.shell_error == 0, out
+end
+
+--- Run a system command and timeout after 30 seconds.
+---
+--- @param cmd table List of command arguments to execute
+--- @param args ?table Optional arguments:
+---                   - stdin (string): Data to write to the job's stdin
+---                   - stderr (boolean): Append stderr to stdout
+---                   - ignore_error (boolean): If true, ignore error output
+---                   - timeout (number): Number of seconds to wait before timing out (default 30)
+function M.system(cmd, args)
+  args = args or {}
+  local stdin = args.stdin or ''
+  local stderr = vim.F.if_nil(args.stderr, false)
+  local ignore_error = vim.F.if_nil(args.ignore_error, false)
+
+  local shell_error_code = 0
+  local opts = {
+    add_stderr_to_output = stderr,
+    output = '',
+    stderr = '',
+    on_stdout = system_handler,
+    on_stderr = system_handler,
+    on_exit = function(_, data)
+      shell_error_code = data
+    end,
+  }
+  local jobid = vim.fn.jobstart(cmd, opts)
+
+  if jobid < 1 then
+    local message =
+        string.format('Command error (job=%d): %s (in %s)', jobid, shellify(cmd), vim.loop.cwd())
+    error(message)
+    return opts.output, 1
+  end
+
+  if stdin:find('^%s$') then
+    vim.fn.jobsend(jobid, stdin)
+  end
+
+  local res = vim.fn.jobwait({ jobid }, vim.F.if_nil(args.timeout, 30) * 1000)
+  if res[1] == -1 then
+    error('Command timed out: ' .. shellify(cmd))
+    vim.fn.jobstop(jobid)
+  elseif shell_error_code ~= 0 and not ignore_error then
+    local emsg = string.format(
+      'Command error (job=%d, exit code %d): %s (in %s)',
+      jobid,
+      shell_error_code,
+      shellify(cmd),
+      vim.loop.cwd()
+    )
+    if opts.output:find('%S') then
+      emsg = string.format('%s\noutput: %s', emsg, opts.output)
+    end
+    if opts.stderr:find('%S') then
+      emsg = string.format('%s\nstderr: %s', emsg, opts.stderr)
+    end
+    error(emsg)
+  end
+
+  -- return opts.output
+  return vim.trim(vim.fn.system(cmd)), shell_error_code
```

Comment:
:eyes:

Reply:
Why the googly eyes?

Reply:
The job is started with jobstart(), but ignores the output from it, and then re-runs the command with system() to get the output. Anyways, looks like #26848 addresses this.